addons/sale_project/models/project.py

```diff
@@ -45,6 +47,27 @@ def _compute_has_any_so_to_invoice(self):
                     has_any_so_to_invoice = has_any_so_to_invoice or (task.sale_order_id and task.sale_order_id.invoice_status == "to invoice")
                 project.has_any_so_to_invoice = has_any_so_to_invoice
 
+    @api.depends('sale_order_id')
+    def _compute_display_create_invoice_buttons(self):
+        primary, secondary = False, False
+        if not self.user_has_groups('sales_team.group_sale_salesman_all_leads'):
+            self.update({
+                'display_create_invoice_primary': primary,
+                'display_create_invoice_secondary': secondary,
+            })
+        else:
+            for project in self:
+                if project.has_any_so_to_invoice:
+                    primary, secondary = True, False
+                else:
+                    secondary = project.sale_order_id and project.sale_order_id.invoice_status == "no"
+                    for task in project.task_ids:
+                        secondary = secondary or (task.sale_order_id and task.sale_order_id.invoice_status == "no")
```

Comment:
Duplicate code from compute method of `has_any_so_to_invoice` ? 🤔

Reply:
Well, not really, as in this case we search for invoice_status == "no" instead of "to_invoice". If the project or its tasks have an invoice status of "to_invoice", we want to display the button in primary ; otherwise, if the invoice status is "no", we want to display it in secondary. If not, then we simply don't display the button.